broker-daemon/block-order-worker/index.js

```diff
@@ -302,36 +302,93 @@ class BlockOrderWorker extends EventEmitter {
+   * @returns {Big} res.outbound
+   */
+  async activeAmountsForOrders (orders) {
+    const response = {
+      inbound: Big(0),
+      outbound: Big(0)
+    }
+    return orders.reduce((acc, { order, state }) => {
+      // If the order is not in an active state , then we can simply skip it
+      if (!Object.values(OrderStateMachine.ACTIVE_STATES).includes(state)) {
+        return acc
+      }
+
+      if (state === OrderStateMachine.STATES.EXECUTING) {
+        acc.inbound = acc.inbound.plus(order.inboundFillAmount)
+        acc.outbound = acc.outbound.plus(order.outboundFillAmount)
+        return acc
+      } else {
+        acc.inbound = acc.inbound.plus(order.inboundAmount)
+        acc.outbound = acc.outbound.plus(order.outboundAmount)
+        return acc
+      }
+    }, response)
+  }
+
+  /**
+   * Calculates active amounts for a given collection of fills
+   * @param {Array<Fill>} fills
+   * @returns {Object} res
+   * @returns {Big} res.inbound
+   * @returns {Big} res.outbound
+   */
+  async activeAmountsForFills (fills) {
+    const response = {
+      inbound: Big(0),
+      outbound: Big(0)
+    }
+    return fills.reduce((acc, { fill, state }) => {
+      // If the fill is not in an active state , then we can simply skip it
+      if (!Object.values(FillStateMachine.ACTIVE_STATES).includes(state)) {
+        return acc
+      }
+
+      acc.inbound = acc.inbound.plus(fill.inboundAmount)
+      acc.outbound = acc.outbound.plus(fill.outboundAmount)
+      return acc
+    }, response)
+  }
+
   /**
    * Adds up active/committed funds in inbound and outbound orders/fills
    *
-   * @param {string} marketName  - Name of the market to creat the block order in (e.g. BTC/LTC)
-   * @param {string} side        - Side of the market to take (e.g. BID or ASK)
-   * @returns {Object} contains activeOutboundAmount and activeInboundAmount of orders/fills
+   * @todo Change return value from Big to String
+   * @param {string} side
+   * @returns {Object} res
+   * @returns {Big} activeOutboundAmount
+   * @returns {Big} activeInboundAmount
    */
-  async calculateActiveFunds (marketName, side) {
-    const blockOrders = await this.getBlockOrders(marketName)
-    const blockOrdersForSide = blockOrders.filter(blockOrder => blockOrder.side === side)
-
-    const fullBlockOrders = await Promise.all(
-      blockOrdersForSide.map(async (bo) => {
-        await Promise.all([
-          bo.populateOrders(this.ordersStore),
-          bo.populateFills(this.fillsStore)
-        ])
-        return bo
-      })
-    )
+  async calculateActiveFunds (side) {
+    let orders = await Order.getAllOrders(this.ordersStore)
+    let fills = await Fill.getAllFills(this.fillsStore)
+
+    // Filter out records for a particular side if it exists. This is needed to be able
+    // to calculate inbound/outbound amounts for the network status tab.
+    //
+    // TODO: Allow calculateActiveFunds to use market by indexes on orders/fills
+    if (side) {
```

Comment:
is it impossible to filter based on market for fills and orders? (is that why the TODO is here?)

Reply:
nope. we could create a custom method for it, but yeah, that is why I left the TODO. Most of the tradeoffs I made were for performance. if we filter all of those records, it just increases the time for the calls.